### PR TITLE
Web Hardening by Stas Oct 4

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js [hunk 1]
```diff
@@ -103,13 +111,93 @@ export default class TokenStep extends React.PureComponent {
...
+    event.preventDefault();
+    this.setState({ selection: 'generate' });
+  };
+
+  handleUseExistingClick = (event /*: Event */) => {
+    event.preventDefault();
+    this.setState({ selection: 'use-existing' });
+  };
+
+  handleExisingTokenChange = (event /*: { currentTarget: HTMLInputElement } */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js [hunk 2]
```diff
@@ -103,13 +111,93 @@ export default class TokenStep extends React.PureComponent {
...
     event.preventDefault();
-    if (this.state.token) {
-      this.props.onContinue(this.state.token);
+    const token = this.getToken();
+    if (token) {
+      this.props.onContinue(token);
     }
   };
 
+  handleGenerateClick = (event /*: Event */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js [hunk 3]
```diff
@@ -103,13 +111,93 @@ export default class TokenStep extends React.PureComponent {
...
+    event.preventDefault();
+    this.setState({ selection: 'generate' });
+  };
+
+  handleUseExistingClick = (event /*: Event */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js [hunk 4]
```diff
@@ -103,13 +111,93 @@ export default class TokenStep extends React.PureComponent {
...
+              onChange={this.handleTokenNameChange}
+              placeholder={translate('onboading.token.generate_token.placeholder')}
+              required={true}
+              type="text"
+              value={this.state.tokenName || ''}
+            />
+            {this.state.loading ? (
+              <i className="spinner text-middle" />
+            ) : (
+              <button className="text-middle">{translate('onboarding.token.generate')}</button>
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js]: I think this button should be disabled as long as the token name is empty

File: server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js [hunk 5]
```diff
@@ -161,7 +238,8 @@ export default class TokenStep extends React.PureComponent {
   };
 
   renderResult = () => {
-    const { token, tokenName } = this.state;
+    const { selection, tokenName } = this.state;
+    const token = this.getToken();
 
     if (!token) {
```
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/tutorials/onboarding/TokenStep.js]: I'd check for `selection === 'generate'` here. I don't think we ever want to render this in "existing" mode.

**Pull request merged**